stacks_queues/queue.py

```diff
@@ -16,47 +16,90 @@ def __init__(self):
         self.rear = -1
         self.size = 0
       
-
+    # put something in back and comes off through the front
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
-
+        # checks to see if the list is empty
+        # if list is empty we want to add the element to the list set it to front
+        if self.front == -1 and self.rear == -1:
+            self.front += 1
+            self.store[self.front] = element
+        
+        elif self.front == (self.rear + 1) % INITIAL_QUEUE_SIZE:
+            raise
+        
+        else:
+            self.rear = (self.rear + 1) % INITIAL_QUEUE_SIZE
+            self.store[self.rear] = element 
+        
+        
+        
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.front == -1 and self.rear == -1:
+            return None
+        else:
+            self.store[self.front + 1] = self.front
 
+    
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.front == -1 and self.rear == -1: 
+            return None
+        else:
+            return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        length = 0
+        for position in INITIAL_QUEUE_SIZE:
+            if position:
+                length += 1
+        return length 
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.front == -1 and self.rear == -1:
+            return True
+        return False
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        queue = []
+        starting_index = self.store[self.front]
```

Comment:
This is going to take the value of `self.store[self.front]` and use the value as an index.